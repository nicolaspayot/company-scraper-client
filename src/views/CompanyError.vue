<template>
  <Card :title="'Whoops'" :button-title="'New search'" @submit="next()">
    <div class="error">
      <h2 v-if="error.statusCode === 404">
        <i class="fas fa-search"></i>
        <span class="error__txt">Sorry, we couldn't find anything.</span>
      </h2>
      <h2 v-else>
        <i class="fas fa-exclamation-triangle"></i>
        <span class="error__txt">Sorry, something went wrong.</span>
      </h2>
    </div>
  </Card>
</template>

<script>
import { mapState } from 'vuex';
import Card from '../components/Card.vue';

export default {
  name: 'Error',
  components: {
    Card,
  },
  computed: {
    ...mapState(['error']),
  },
  methods: {
    next() {
      this.$router.push({ name: 'CompanySearch' });
    },
  },
};
</script>

<style lang="scss" scoped>
.error {
  display: flex;
  justify-content: center;
}

.error__txt {
  margin-left: 10px;
}
</style>
