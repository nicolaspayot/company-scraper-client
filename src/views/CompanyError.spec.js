import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import CompanyError from './CompanyError.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('CompanyError.vue', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = new Vuex.Store({
      state: { error: { statusCode: 404 } },
    });
    wrapper = shallowMount(CompanyError, { store, localVue });
  });

  it('should render no result error', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render server error', () => {
    store.state.error = { stateCode: 500 };
    expect(wrapper.element).toMatchSnapshot();
  });
});
