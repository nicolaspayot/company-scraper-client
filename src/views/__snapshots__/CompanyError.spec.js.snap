// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyError.vue should render no result error 1`] = `
<card-stub
  buttontitle="New search"
  title="Whoops"
>
  <div
    class="error"
  >
    <h2>
      <i
        class="fas fa-search"
      />
       
      <span
        class="error__txt"
      >
        Sorry, we couldn't find anything.
      </span>
    </h2>
  </div>
</card-stub>
`;

exports[`CompanyError.vue should render server error 1`] = `
<card-stub
  buttontitle="New search"
  title="Whoops"
>
  <div
    class="error"
  >
    <h2>
      <i
        class="fas fa-exclamation-triangle"
      />
       
      <span
        class="error__txt"
      >
        Sorry, something went wrong.
      </span>
    </h2>
  </div>
</card-stub>
`;
